apps/src/applab/applab.js

```diff
@@ -3169,96 +3172,120 @@ Applab.handleError = function(errorCallback, message) {
-  apiValidateType(opts, 'getKeyValue', 'onSuccess', opts.onSuccess, 'function');
+  apiValidateType(opts, 'getKeyValue', 'callback', opts.onSuccess, 'function');
   apiValidateType(opts, 'getKeyValue', 'onError', opts.onError, 'function', OPTIONAL);
-  var onSuccess = Applab.handleReadValue.bind(this, opts.onSuccess);
-  var onError = Applab.handleError.bind(this, opts.onError);
+  opts.interpreter = Applab.interpreter;
+  var onSuccess = Applab.handleReadValue.bind(this, opts);
+  var onError = Applab.handleError.bind(this, opts);
   AppStorage.getKeyValue(opts.key, onSuccess, onError);
 };
 
-Applab.handleReadValue = function(successCallback, value) {
-  if (successCallback) {
+Applab.handleReadValue = function(opts, value) {
+  // Ensure that this event was requested by the same instance of the interpreter
+  // that is currently active before proceeding...
+  if (opts.onSuccess && opts.interpreter === Applab.interpreter) {
     Applab.eventQueue.push({
-      'fn': successCallback,
+      'fn': opts.onSuccess,
       'arguments': [value]
     });
   }
 };
 
 Applab.setKeyValue = function(opts) {
+  // PARAMNAME: setKeyValue: callback vs. callbackFunction
   apiValidateType(opts, 'setKeyValue', 'key', opts.key, 'string');
   apiValidateType(opts, 'setKeyValue', 'value', opts.value, 'primitive');
-  apiValidateType(opts, 'setKeyValue', 'onSuccess', opts.onSuccess, 'function', OPTIONAL);
+  apiValidateType(opts, 'setKeyValue', 'callback', opts.onSuccess, 'function', OPTIONAL);
   apiValidateType(opts, 'setKeyValue', 'onError', opts.onError, 'function', OPTIONAL);
-  var onSuccess = Applab.handleSetKeyValue.bind(this, opts.onSuccess);
-  var onError = Applab.handleError.bind(this, opts.onError);
+  opts.interpreter = Applab.interpreter;
+  var onSuccess = Applab.handleSetKeyValue.bind(this, opts);
+  var onError = Applab.handleError.bind(this, opts);
   AppStorage.setKeyValue(opts.key, opts.value, onSuccess, onError);
 };
 
-Applab.handleSetKeyValue = function(successCallback) {
-  if (successCallback) {
+Applab.handleSetKeyValue = function(opts) {
+  // Ensure that this event was requested by the same instance of the interpreter
+  // that is currently active before proceeding...
+  if (opts.onSuccess && opts.interpreter === Applab.interpreter) {
     Applab.eventQueue.push({
-      'fn': successCallback,
+      'fn': opts.onSuccess,
       'arguments': []
     });
   }
 };
 
 Applab.readRecords = function (opts) {
+  // PARAMNAME: readRecords: table vs. tableName
+  // PARAMNAME: readRecords: callback vs. callbackFunction
+  // PARAMNAME: readRecords: terms vs. searchTerms
   apiValidateType(opts, 'readRecords', 'table', opts.table, 'string');
-  apiValidateType(opts, 'readRecords', 'searchParams', opts.searchParams, 'object');
-  apiValidateType(opts, 'readRecords', 'onSuccess', opts.onSuccess, 'function');
+  apiValidateType(opts, 'readRecords', 'searchTerms', opts.searchParams, 'object');
```

Comment:
can you explain why you are adding this param and calling it 'terms' in one place and 'searchTerms' in the other? I get that this is flagged to resolve later, but since this is an inconsistency that you're introducing in this PR I am worried I might not understand what the plan is here.